pages/learn/develop/container-contracts.md

```diff
@@ -0,0 +1,112 @@
+---
+title: Container contracts
+excerpt: A guide to using container contracts to enforce device compatibility
+---
+
+# Container contracts
+
+**Note** Container contracts are available for devices running Supervisor versions >= 10.6.17. All prior versions will not enforce any contracts.
+
+Container contracts are used to ensure the compatibility of a device to run an individual service of an application. For example, container contracts may be used to ensure that the device is running a specific version of [{{ $names.company.lower }} Supervisor][supervisor] or has a specific version of the [NVIDIA Tegra Linux Driver Package][l4t] (L4T).
```

Comment:
For somebody outside of the Balena world, "compatibility of a device" implies hardware compatibility to me, but I guess you're actually talking about software components compatibility?

Reply:
I suppose it's a bit of both given L4T can only run on certain devices but point duly noted. I've left as is for now.